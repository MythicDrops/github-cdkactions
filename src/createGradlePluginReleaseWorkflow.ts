import { Job, Stack, Workflow } from "cdkactions";

export const createGradlePluginReleaseWorkflow = (stack: Stack): Workflow => {
  const workflow = new Workflow(stack, "release", {
    name: "Create Release",
    on: {
      push: { tags: ["*"] },
    },
  });
  new Job(workflow, "release", {
    name: "Release",
    runsOn: "ubuntu-latest",
    steps: [
      {
        name: "Check-out Code",
        uses: "actions/checkout@v2",
        with: { "fetch-depth": "0" },
      },
      {
        name: "Build Changelog",
        id: "github_release",
        uses: "mikepenz/release-changelog-builder-action@main",
        env: {
          GITHUB_TOKEN: `\${{ secrets.GITHUB_TOKEN }}`,
        },
      },
      {
        name: "Create GitHub Release",
        uses: "actions/create-release@v1",
        env: {
          GITHUB_TOKEN: `\${{ secrets.GITHUB_TOKEN }}`,
        },
        with: {
          body: `\${{ steps.github_release.outputs.changelog }}`,
          release_name: `\${{ github.ref }}`,
          tag_name: `\${{ github.ref }}`,
        },
      },
    ],
  });
  return workflow;
};
