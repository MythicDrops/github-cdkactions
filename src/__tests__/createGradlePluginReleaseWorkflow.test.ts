import { App, Stack } from "cdkactions";
import { createGradlePluginReleaseWorkflow } from "../createGradlePluginReleaseWorkflow";

describe("createGradlePluginReleaseWorkflow", () => {
  let app: App;
  let stack: Stack;

  beforeEach(() => {
    app = new App();
    stack = new Stack(app, "cdk");
  });

  it("generates a GitHub Actions workflow", () => {
    expect(
      createGradlePluginReleaseWorkflow(stack).toGHAction()
    ).toMatchSnapshot();
  });
});
