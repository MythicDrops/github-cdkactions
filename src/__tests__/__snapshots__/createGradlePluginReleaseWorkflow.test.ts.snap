// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGradlePluginReleaseWorkflow generates a GitHub Actions workflow 1`] = `
Object {
  "jobs": Object {
    "release": Object {
      "name": "Release",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "name": "Check-out Code",
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
          },
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "id": "github_release",
          "name": "Build Changelog",
          "uses": "mikepenz/release-changelog-builder-action@main",
        },
        Object {
          "env": Object {
            "GITHUB_TOKEN": "\${{ secrets.GITHUB_TOKEN }}",
          },
          "name": "Create GitHub Release",
          "uses": "actions/create-release@v1",
          "with": Object {
            "body": "\${{ steps.github_release.outputs.changelog }}",
            "release_name": "\${{ github.ref }}",
            "tag_name": "\${{ github.ref }}",
          },
        },
      ],
    },
  },
  "name": "Create Release",
  "on": Object {
    "push": Object {
      "tags": Array [
        "*",
      ],
    },
  },
}
`;
